### DISCUSSION TITLE
Use ASGI event types from Uvicorn

### DISCUSSION THREAD

**Discussion by Masynchin** [encode/starlette#2652] [Ideas] [answered]
As for now, `starlette/types.py` offers such types for ASGI events:

https://github.com/encode/starlette/blob/07427f86474b15d648497014eb19c41f01317f15/starlette/types.py#L10-L11

Unlike the rich types from `uvicorn/_types.py`:

https://github.com/encode/uvicorn/blob/5bf788f0eb0fc771f5c4eed8f282c7ec256565d2/uvicorn/_types.py#L68-L82

Is it a good idea to port this types from `uvicorn` here?

**Comment by Kludex** [comment, marked as answer]
There were some attempts to do that before. It's too much work... But you are welcome to create a PR.

**Reply by Masynchin** [reply]
> There were some attempts to do that before. It's too much work... But you are welcome to create a PR.

I decided that I do not give enough time for the implementation, but can provide a little more to the conversation. Can we use [your library](https://github.com/Kludex/asgi-types) for the types, like the optional dependency, and fallback to the current ones when it is not installed?
